types/react/index.d.ts

```diff
@@ -186,7 +186,7 @@ declare namespace React {
     // DOM Elements
     function createElement<P extends HTMLAttributes<T>, T extends HTMLElement>(
         type: keyof ReactHTML,
-        props?: ClassAttributes<T> & P,
+        props?: InputHTMLAttributes<T> & ClassAttributes<T> & P,
```

Comment:
this isn't quite right—I meant something like:

```ts
function createElement(
    type: "input",
    props?: InputHTMLAttributes<HTMLInputElement> & ClassAttributes<HTMLInputElement>,
    ...
```

as an _additional_ overload for `createElement`.

If we're thinking more generally, ideally there would be an interface that maps each DOM type to its attributes interface, like

```ts
interface DOMAttributesMap {
    input: InputHTMLAttributes<HTMLInputElement> & ClassAttributes<HTMLInputElement>,
    ...
}
```

and then we can type `createElement` with lookup types:

```ts
function createElement<T extends keyof DOMAttributesMap>(
    type: T,
    props?: DOMAttributesMap[T],
    ...
```

but we don't have to tackle that in this PR.